### ISSUE TITLE
ERROR: /data/Installers/tensorflow/tensorflow/core/kernels/BUILD:2602:1: C++ compilation of rule '//tensorflow/core/kernels:cwise_op' failed (Exit 1): computecpp failed

### ISSUE THREAD

**Issue by znmeb** [lukeiwanski/tensorflow#193]
```
ERROR: /data/Installers/tensorflow/tensorflow/core/kernels/BUILD:2602:1: C++ compilation of rule '//tensorflow/core/kernels:cwise_op' failed (Exit 1): computecpp failed: error executing command
  (cd /home/znmeb/.cache/bazel/_bazel_znmeb/47854f00488658b16693bf4601e6f0fd/execroot/org_tensorflow && \
  exec env - \
    COMPUTECPP_TOOLKIT_PATH=/usr/local/computecpp \
    HOST_CXX_COMPILER=/usr/bin/g++ \
    HOST_C_COMPILER=/usr/bin/gcc \
    LD_LIBRARY_PATH=/usr/local/computecpp/lib \
    PATH=/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/znmeb/miniconda3/bin \
    PWD=/proc/self/cwd \
    PYTHON_BIN_PATH=/usr/bin/python \
    PYTHON_LIB_PATH=/usr/lib/python3.6/site-packages \
    TF_NEED_COMPUTECPP=1 \
    TF_NEED_CUDA=0 \
    TF_NEED_OPENCL_SYCL=1 \
```

Full build log is at <https://gist.github.com/znmeb/0d003c1eb496793dcca34a4f0ef4b293>

Notes:
1. This succeeded a couple of days ago. System is Arch Linux.
2. Branch is `dev/amd_gpu`.
3. First page of `git log`:
```
commit e5137e1a58d8c61e0f6ec718d4e2a970c209d988 (HEAD -> dev/amd_gpu, origin/dev/amd_gpu)
Author: Romain Biessy <romain.biessy@codeplay.com>
Date:   Mon Dec 18 15:08:38 2017 +0000

    [SYCL] Add EIGEN_HAS_CXX11_MATH flag (#191)

commit 6631d9c054c9b0531e8f9fd89031bb7f0c4612e1
Author: Luke Iwanski <luke@codeplay.com>
Date:   Fri Dec 15 21:53:31 2017 +0000

    Fixes https://github.com/lukeiwanski/tensorflow/issues/189

commit e8852d3eae9dc357fb7e2f0f68b5f9766baf1371
Author: Luke Iwanski <luke@codeplay.com>
Date:   Wed Dec 13 11:33:40 2017 +0000

    Fixed typo

commit c8e23508a409b24d026d0b7efc2cf0d9d08a96fa
Author: Luke Iwanski <luke@codeplay.com>
Date:   Wed Dec 13 04:35:33 2017 +0000

    [SYCL][Docker] Introducing Dockerfile for AMD OpenCL on Ubuntu16.04
```

**Comment by lukeiwanski** [comment]
Hi @znmeb thanks for the report,

Could you provide the log after the error? ( there should be some sort of compilation error either from computecpp or your host compiler ) 
The gist you provided has a lot of warnings but no actual error.

**Comment by znmeb** [comment]
@lukeiwanski The error's not in the gist? I thought I had the whole logfile. The bazel command is

```
bazel build --config=opt --config=sycl --verbose_failures \
  //tensorflow/tools/pip_package:build_pip_package 2>&1 | tee $HERE/tfbuild.log
```

and the gist is $HERE/tfbuild.log.

Try https://gist.githubusercontent.com/znmeb/0d003c1eb496793dcca34a4f0ef4b293/raw/9ebf59266f4ad8c330b472d0471bd1185fc89995/tfbuild.log

and do a save page.

**Comment by znmeb** [comment]
I *think* this is the whole thing - they weren't kidding when they said "verbose":

```
[5,939 / 6,008] Compiling tensorflow/core/kernels/argmax_op.cc; 136s local ... (8 actions running)
ERROR: /data/Installers/tensorflow/tensorflow/core/kernels/BUILD:2602:1: C++ compilation of rule '//tensorflow/core/kernels:cwise_op' failed (Exit 1): computecpp failed: error executing command 
  (cd /home/znmeb/.cache/bazel/_bazel_znmeb/47854f00488658b16693bf4601e6f0fd/execroot/org_tensorflow && \
  exec env - \
    COMPUTECPP_TOOLKIT_PATH=/usr/local/computecpp \
    HOST_CXX_COMPILER=/usr/bin/g++ \
    HOST_C_COMPILER=/usr/bin/gcc \
    LD_LIBRARY_PATH=/usr/local/computecpp/lib \
    PATH=/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/znmeb/miniconda3/bin \
    PWD=/proc/self/cwd \
    PYTHON_BIN_PATH=/usr/bin/python \
    PYTHON_LIB_PATH=/usr/lib/python3.6/site-packages \
    TF_NEED_COMPUTECPP=1 \
    TF_NEED_CUDA=0 \
    TF_NEED_OPENCL_SYCL=1 \
  external/local_config_sycl/crosstool/computecpp -fPIE -fno-omit-frame-pointer -Wall -g0 -O2 -DNDEBUG -ffunction-sections -fdata-sections -DGEMMLOWP_ALLOW_SLOW_SCALAR_FALLBACK '-march=native' '-std=c++11' -MD -MF bazel-out/k8-py3-opt/bin/tensorflow/core/kernels/_objs/cwise_op/tensorflow/core/kernels/cwise_op_erf.pic.d '-frandom-seed=bazel-out/k8-py3-opt/bin/tensorflow/core/kernels/_objs/cwise_op/tensorflow/core/kernels/cwise_op_erf.pic.o' -fPIC -DEIGEN_MPL2_ONLY -D__CLANG_SUPPORT_DYN_ANNOTATION__ -DTENSORFLOW_USE_JEMALLOC -DTENSORFLOW_USE_ABSL -DTF_USE_SNAPPY -DPLATFORM_LINUX -DENABLE_CURL_CLIENT -DENABLE_NO_ENCRYPTION -iquote . -iquote bazel-out/k8-py3-opt/genfiles -iquote external/nsync -iquote bazel-out/k8-py3-opt/genfiles/external/nsync -iquote external/bazel_tools -iquote bazel-out/k8-py3-opt/genfiles/external/bazel_tools -iquote external/eigen_archive -iquote bazel-out/k8-py3-opt/genfiles/external/eigen_archive -iquote external/local_config_sycl -iquote bazel-out/k8-py3-opt/genfiles/external/local_config_sycl -iquote external/com_google_absl -iquote bazel-out/k8-py3-opt/genfiles/external/com_google_absl -iquote external/jemalloc -iquote bazel-out/k8-py3-opt/genfiles/external/jemalloc -iquote external/gif_archive -iquote bazel-out/k8-py3-opt/genfiles/external/gif_archive -iquote external/jpeg -iquote bazel-out/k8-py3-opt/genfiles/external/jpeg -iquote external/protobuf_archive -iquote bazel-out/k8-py3-opt/genfiles/external/protobuf_archive -iquote external/com_googlesource_code_re2 -iquote bazel-out/k8-py3-opt/genfiles/external/com_googlesource_code_re2 -iquote external/farmhash_archive -iquote bazel-out/k8-py3-opt/genfiles/external/farmhash_archive -iquote external/fft2d -iquote bazel-out/k8-py3-opt/genfiles/external/fft2d -iquote external/highwayhash -iquote bazel-out/k8-py3-opt/genfiles/external/highwayhash -iquote external/png_archive -iquote bazel-out/k8-py3-opt/genfiles/external/png_archive -iquote external/zlib_archive -iquote bazel-out/k8-py3-opt/genfiles/external/zlib_archive -iquote external/curl -iquote bazel-out/k8-py3-opt/genfiles/external/curl -iquote external/boringssl -iquote bazel-out/k8-py3-opt/genfiles/external/boringssl -iquote external/jsoncpp_git -iquote bazel-out/k8-py3-opt/genfiles/external/jsoncpp_git -iquote external/aws -iquote bazel-out/k8-py3-opt/genfiles/external/aws -isystem external/nsync/public -isystem bazel-out/k8-py3-opt/genfiles/external/nsync/public -isystem external/bazel_tools/tools/cpp/gcc3 -isystem external/eigen_archive -isystem bazel-out/k8-py3-opt/genfiles/external/eigen_archive -isystem external/local_config_sycl/sycl -isystem bazel-out/k8-py3-opt/genfiles/external/local_config_sycl/sycl -isystem external/local_config_sycl/sycl/include -isystem bazel-out/k8-py3-opt/genfiles/external/local_config_sycl/sycl/include -isystem external/jemalloc/include -isystem bazel-out/k8-py3-opt/genfiles/external/jemalloc/include -isystem external/gif_archive/lib -isystem bazel-out/k8-py3-opt/genfiles/external/gif_archive/lib -isystem external/protobuf_archive/src -isystem bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src -isystem external/farmhash_archive/src -isystem bazel-out/k8-py3-opt/genfiles/external/farmhash_archive/src -isystem external/png_archive -isystem bazel-out/k8-py3-opt/genfiles/external/png_archive -isystem external/zlib_archive -isystem bazel-out/k8-py3-opt/genfiles/external/zlib_archive -isystem external/curl/include -isystem bazel-out/k8-py3-opt/genfiles/external/curl/include -isystem external/boringssl/src/include -isystem bazel-out/k8-py3-opt/genfiles/external/boringssl/src/include -isystem external/jsoncpp_git/include -isystem bazel-out/k8-py3-opt/genfiles/external/jsoncpp_git/include -isystem external/aws/aws-cpp-sdk-core/include -isystem bazel-out/k8-py3-opt/genfiles/external/aws/aws-cpp-sdk-core/include -isystem external/aws/aws-cpp-sdk-s3/include -isystem bazel-out/k8-py3-opt/genfiles/external/aws/aws-cpp-sdk-s3/include -DEIGEN_AVOID_STL_ARRAY -Iexternal/gemmlowp -Wno-sign-compare -fno-exceptions '-ftemplate-depth=900' -msse3 -pthread -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c tensorflow/core/kernels/cwise_op_erf.cc -o bazel-out/k8-py3-opt/bin/tensorflow/core/kernels/_objs/cwise_op/tensorflow/core/kernels/cwise_op_erf.pic.o)
/usr/local/computecpp/bin/compute++: /usr/lib/libtinfo.so.5: no version information available (required by /usr/local/computecpp/bin/compute++)
/usr/local/computecpp/bin/compute++: /usr/lib/libtinfo.so.5: no version information available (required by /usr/local/computecpp/bin/compute++)
/usr/local/computecpp/bin/compute++: /usr/lib/libtinfo.so.5: no version information available (required by /usr/local/computecpp/bin/compute++)
/usr/local/computecpp/bin/compute++: /usr/lib/libtinfo.so.5: no version information available (required by /usr/local/computecpp/bin/compute++)
In file included from <built-in>:1:
In file included from ./tensorflow/core/kernels/cwise_op_erf.cc:16:
In file included from ./tensorflow/core/kernels/cwise_ops_common.h:24:
In file included from ./tensorflow/core/kernels/cwise_ops_sycl_common.h:23:
In file included from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:32:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/../SpecialFunctions:46:
external/eigen_archive/unsupported/Eigen/CXX11/../src/SpecialFunctions/SpecialFunctionsImpl.h:342:5: error: static_assert failed "THIS_TYPE_IS_NOT_SUPPORTED"
    EIGEN_STATIC_ASSERT((internal::is_same<Scalar, Scalar>::value == false),
    ^                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/StaticAssert.h:32:40: note: expanded from macro 'EIGEN_STATIC_ASSERT'
    #define EIGEN_STATIC_ASSERT(X,MSG) static_assert(X,#MSG);
                                       ^             ~
```

**Comment by lukeiwanski** [comment]
Yes verbose is quite.. that :) 

I fixed this issue: https://github.com/lukeiwanski/tensorflow/commit/24edbf67bc875db5d67882b82517803c0e3d20cc

can you give it a spin now?

**Comment by znmeb** [comment]
The build completed successfully! I'm going to run it on CIFAR10 now.

It's working! This GPU peaks at something like 1.7 32-bit TFLOPS - the times are about half of what stock TensorFlow 1.4.0 got last night (fully optimized with MKL) on my 8-core 4 GHz AMD processor.
```
python cifar10_train.py
Filling queue with 20000 CIFAR images before starting to train. This will take a few minutes.
2017-12-19 13:34:58.596685: I ./tensorflow/core/common_runtime/sycl/sycl_device.h:70] Found following OpenCL devices:
2017-12-19 13:34:58.596727: I ./tensorflow/core/common_runtime/sycl/sycl_device.h:72] id: 0, type: GPU, name: Bonaire, vendor: Advanced Micro Devices, Inc., profile: FULL_PROFILE
2017-12-19 13:40:47.869761: step 0, loss = 4.68 (3.7 examples/sec; 34.954 sec/batch)
2017-12-19 13:40:53.195375: step 10, loss = 4.61 (240.3 examples/sec; 0.533 sec/batch)
2017-12-19 13:40:58.432293: step 20, loss = 4.58 (244.4 examples/sec; 0.524 sec/batch)
2017-12-19 13:41:03.659911: step 30, loss = 4.52 (244.9 examples/sec; 0.523 sec/batch)
2017-12-19 13:41:08.897797: step 40, loss = 4.31 (244.4 examples/sec; 0.524 sec/batch)
2017-12-19 13:41:14.157540: step 50, loss = 4.34 (243.4 examples/sec; 0.526 sec/batch)
2017-12-19 13:41:19.438207: step 60, loss = 4.27 (242.4 examples/sec; 0.528 sec/batch)
2017-12-19 13:41:24.663681: step 70, loss = 4.58 (245.0 examples/sec; 0.523 sec/batch)
```

**Comment by znmeb** [comment]
![screenshot from 2017-12-19 14-38-13](https://user-images.githubusercontent.com/4938/34182273-68d16144-e4ca-11e7-994f-8b675d823751.png)

Screenshot of `radeontop` during cifar10 training - this is the steady-state (80% of graphics pipe and shader interpolator, 50% of texture addresser). I'm impressed.
Python is using about 2 cores' worth of CPU time. 

But the loss is decreasing very slowly - *that* I am not impressed with! ;-)

**Comment by huoyushi** [comment]
Hello, I got a error with the same build command as yours. The command is 
bazel build --config=opt --config=sycl --verbose_failures //tensorflow/tools/pip_package:build_pip_package 2>&1 | tee ~/build.log     
with the option --config=opt I got the error about protobuf

ERROR: /home/huoyushi/.cache/bazel/_bazel_huoyushi/0e0c71aaae1b5fda4e6d41693b9a0952/external/protobuf_archive/BUILD:656:1: undeclared inclusion(s) in rule '@protobuf_archive//:python/google/protobuf/internal/_api_implementation.so':
this rule is missing dependency declarations for the following files included by 'external/protobuf_archive/python/google/protobuf/internal/api_implementation.cc':

bazel verison 0.8.0 . ubuntu 16.04    gcc 5.4.0  can you help  @znmeb

**Comment by znmeb** [comment]
@huoyushi I can't help but I suspect @lukeiwanski can ;-) What's your default Python? I used 3.6.3 from Arch; Ubuntu's is probably older. If it's trying to install something in the `root` Python area it not only won't work but can break your system if you give it `root` privileges!

**Comment by lukeiwanski** [comment]
@huoyushi can you share exact fail message?

**Comment by huoyushi** [comment]
@lukeiwanski 
I use anaconda create a virtual python environment version is 3.6.3  
command:
bazel build --config=opt --config=sycl --verbose_failures //tensorflow/tools/pip_package:build_pip_package 2>&1 | tee ~/tfbuild.log
python 3.6.3 gcc 5.4.0 ubuntu 16.04  bazel 0.9.0  
this is error details
ERROR: /home/huoyushi/.cache/bazel/_bazel_huoyushi/0e0c71aaae1b5fda4e6d41693b9a0952/external/protobuf_archive/BUILD:656:1: undeclared inclusion(s) in rule '@protobuf_archive//:python/google/protobuf/internal/_api_implementation.so':
this rule is missing dependency declarations for the following files included by 'external/protobuf_archive/python/google/protobuf/internal/api_implementation.cc':
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/Python.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/patchlevel.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pyconfig.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pymacconfig.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pyport.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pymacro.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pyatomic.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pymath.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pytime.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/object.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pymem.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/objimpl.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/typeslots.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pyhash.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pydebug.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/bytearrayobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/bytesobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/unicodeobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/longobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/longintrepr.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/boolobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/floatobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/complexobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/rangeobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/memoryobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/tupleobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/listobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/dictobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/odictobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/enumobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/setobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/methodobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/moduleobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/funcobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/classobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/fileobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pycapsule.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/traceback.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pystate.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/sliceobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/cellobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/iterobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/genobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/descrobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/warnings.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/weakrefobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/structseq.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/namespaceobject.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/codecs.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pyerrors.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pyarena.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/modsupport.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pythonrun.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pylifecycle.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/ceval.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/sysmodule.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/osmodule.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/intrcheck.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/import.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/abstract.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/bltinmodule.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/compile.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/code.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/eval.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pyctype.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pystrtod.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pystrcmp.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/dtoa.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/fileutils.h'
  '/home/huoyushi/.conda/envs/tensorflowwithopencl/include/python3.6m/pyfpe.h'
Target //tensorflow/tools/pip_package:build_pip_package failed to build
INFO: Elapsed time: 2719.448s, Critical Path: 54.78s
FAILED: Build did NOT complete successfull

**Comment by znmeb** [comment]
I had some problems building inside a conda environment, so I built using the system Python. You will need "numpy", "pip" and "wheel" installed in your Python stack for the build to work.

**Comment by huoyushi** [comment]
Thanks for help me. I will try with system python @znmeb

**Comment by huoyushi** [comment]
@lukeiwanski I  recurrent the error about protobuf compile with branch dev/amd_gpu. this is detail
command 
bazel build -c opt --config=sycl //tensorflow/tools/pip_package:build_pip_package 2>&1 | tee ~/tfbuild.log
enviroment
ubuntu 16.04 gcc 5.4.0 python 3.6.3  bazel 0.9.0
[1,606 / 1,663] Compiling tensorflow/core/lib/core/threadpool.cc [for host]; 5s local ... (2 actions running)
ERROR: /home/huoyushi/.cache/bazel/_bazel_huoyushi/0e0c71aaae1b5fda4e6d41693b9a0952/external/protobuf_archive/BUILD:656:1: undeclared inclusion(s) in rule '@protobuf_archive//:python/google/protobuf/internal/_api_implementation.so':
this rule is missing dependency declarations for the following files included by 'external/protobuf_archive/python/google/protobuf/internal/api_implementation.cc':
  '/home/huoyushi/anaconda3/include/python3.6m/Python.h'
  '/home/huoyushi/anaconda3/include/python3.6m/patchlevel.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pyconfig.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pymacconfig.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pyport.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pymacro.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pyatomic.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pymath.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pytime.h'
  '/home/huoyushi/anaconda3/include/python3.6m/object.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pymem.h'
  '/home/huoyushi/anaconda3/include/python3.6m/objimpl.h'
  '/home/huoyushi/anaconda3/include/python3.6m/typeslots.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pyhash.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pydebug.h'
  '/home/huoyushi/anaconda3/include/python3.6m/bytearrayobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/bytesobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/unicodeobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/longobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/longintrepr.h'
  '/home/huoyushi/anaconda3/include/python3.6m/boolobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/floatobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/complexobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/rangeobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/memoryobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/tupleobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/listobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/dictobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/odictobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/enumobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/setobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/methodobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/moduleobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/funcobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/classobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/fileobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pycapsule.h'
  '/home/huoyushi/anaconda3/include/python3.6m/traceback.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pystate.h'
  '/home/huoyushi/anaconda3/include/python3.6m/sliceobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/cellobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/iterobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/genobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/descrobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/warnings.h'
  '/home/huoyushi/anaconda3/include/python3.6m/weakrefobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/structseq.h'
  '/home/huoyushi/anaconda3/include/python3.6m/namespaceobject.h'
  '/home/huoyushi/anaconda3/include/python3.6m/codecs.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pyerrors.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pyarena.h'
  '/home/huoyushi/anaconda3/include/python3.6m/modsupport.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pythonrun.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pylifecycle.h'
  '/home/huoyushi/anaconda3/include/python3.6m/ceval.h'
  '/home/huoyushi/anaconda3/include/python3.6m/sysmodule.h'
  '/home/huoyushi/anaconda3/include/python3.6m/osmodule.h'
  '/home/huoyushi/anaconda3/include/python3.6m/intrcheck.h'
  '/home/huoyushi/anaconda3/include/python3.6m/import.h'
  '/home/huoyushi/anaconda3/include/python3.6m/abstract.h'
  '/home/huoyushi/anaconda3/include/python3.6m/bltinmodule.h'
  '/home/huoyushi/anaconda3/include/python3.6m/compile.h'
  '/home/huoyushi/anaconda3/include/python3.6m/code.h'
  '/home/huoyushi/anaconda3/include/python3.6m/eval.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pyctype.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pystrtod.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pystrcmp.h'
  '/home/huoyushi/anaconda3/include/python3.6m/dtoa.h'
  '/home/huoyushi/anaconda3/include/python3.6m/fileutils.h'
  '/home/huoyushi/anaconda3/include/python3.6m/pyfpe.h'
Target //tensorflow/tools/pip_package:build_pip_package failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 2208.207s, Critical Path: 44.15s
FAILED: Build did NOT complete successfully
But I try compiling with master branch from official git 
command
 bazel build -c opt  //tensorflow/tools/pip_package:build_pip_package 2>&1 | tee ~/tfbuild.log
this can build successfully.So can you help me handle the error with option sycl. Thanks anyway @lukeiwanski

**Comment by lukeiwanski** [comment]
@huoyushi thanks for the report. I am looking at this just now. 
btw. That looks like set up/environment/configuration  issue and it is good material for separate issue :)

NOTE: 
I am not able to reproduce your issue at 3e999990f1b69f3c7cd87dc80eb9de48aa61d9b1 can you confirm that this issue still persist on that commit?

The workaround on your system would be to add:

`cxx_builtin_include_directory: "/home/huoyushi/anaconda3/include/"`
to 
`third_party/sycl/crosstool/CROSSTOOL.tpl` around line 52

**Comment by lukeiwanski** [comment]
@znmeb thanks for running benchmarks! :)
It's still work in progress and we will get better

**Comment by huoyushi** [comment]
Thanks a lot @lukeiwanski   I can compile successfully but unfortunately I get an exception about sycl 
this is a common problem on my system ,All  program compiled by computecpp get the same exception :
terminate called after throwing an instance of 'cl::sycl::exception'. 
I will ask DuncanMcBain for help.

**Comment by mirh** [comment]
I guess that kinda makes for this issue solved?
Computecpp issues should go [here](https://github.com/codeplaysoftware/computecpp-sdk/issues).

**Comment by lukeiwanski** [comment]
Closing :)
